Added detailed README with features and setup (#11)
codepvg/leetcode-ranking: README.md

# CodePVG LeetCode Ranking

> A web-based platform developed for CodePVG to track and rank students based on their LeetCode performance.  

Reviewer: ```suggestion
A web-based platform developed by [CodePVG](https://www.linkedin.com/company/codepvg/) to track and rank students of [PVG COET](https://www.pvgcoet.ac.in/) based on their LeetCode performance.
```

It allows users to register with their LeetCode username and automatically fetches their problem-solving statistics to display on a leaderboard.

---

## Purpose

The goal of this project is to:

- Encourage consistent problem-solving among students  
- Create a competitive yet motivating environment  
- Provide visibility into individual coding progress  

---

> ## Features

Reviewer: Let’s avoid including a features section in the README, so we don’t have to keep updating it every time new features are added.

### User Registration
- Users can register using their name and LeetCode username  
- Accepts both direct usernames and profile links  
- Validates whether the LeetCode user exists before registration  

### Leaderboard System
- Displays ranked users based on their LeetCode statistics  
- Automatically updates rankings based on fetched data  
- Provides a clean interface for easy comparison  

### Automated Data Fetching
- Periodically fetches user data using the LeetCode API  
- Updates stored data without manual intervention  
- Keeps leaderboard information up-to-date  

### User Interface
- Simple and clean UI design  
- Smooth user experience with loading states  
- Includes blur and spinner animation during registration  

### Input Handling and Validation
- Extracts username from full LeetCode profile links  
- Prevents invalid or non-existent usernames  
- Avoids duplicate or incorrect entries  

### Error Handling
- Handles API failures gracefully  
- Displays alerts for invalid inputs  
- Prevents multiple submissions during requests  

---

> ## Tech Stack

Reviewer: We already get a summary of the tech stack on the main repo page, so instead of repeating that in the README, we can link the related repos for the backend and the web scraping API (since LeetCode doesn’t officially provide one):

- [leetcode-api](https://github.com/codepvg/leetcode-api)  
- [lc-backend](https://github.com/codepvg/lc-backend)

This would be more useful and avoid redundancy

| Layer     | Technology              |
|----------|------------------------|
| Frontend | HTML, CSS, JavaScript  |
| Backend  | Node.js                |
| API      | LeetCode API           |

---

## Project Structure

```
leetcode-ranking/
│── frontend/        # UI (HTML, CSS, JS)
│── data/            # Stored user data
│── scripts/         # Automation scripts

> │── server.js        # Backend server

Reviewer: ```suggestion
│── server.js        # Express server
```

│── package.json
```

---

## How to Run Locally

### 1. Clone the repository

```bash
git clone https://github.com/YOUR-USERNAME/leetcode-ranking.git
cd leetcode-ranking
```

### 2. Install dependencies

npm install
cd frontend

> npm install

Reviewer: ```suggestion
`npm install`
```

### 3. Run the project

npm run dev

or

> node server.js

Reviewer: ```suggestion
`npm run dev`
or
`npm start`
```

## Usage

1. Open the registration page  
2. Enter your name and LeetCode username  
3. Submit the form  

> 4. View your ranking on the leaderboard  

Reviewer: ```suggestion
4. View your ranking on the leaderboard after the next sync
```

# CodePVG LeetCode Ranking

A web-based platform developed by [CodePVG](https://www.linkedin.com/company/codepvg/) to track and rank students of [PVG COET](https://www.pvgcoet.ac.in/) based on their LeetCode performance

It allows users to register with their LeetCode username and automatically fetches their problem-solving statistics to display on a leaderboard.

---

## Purpose

The goal of this project is to:

- Encourage consistent problem-solving among students  
- Create a competitive yet motivating environment  
- Provide visibility into individual coding progress  

---

This project relies on external services for fetching and processing LeetCode data. 
Instead of duplicating details here, refer to the following repositories:

- [leetcode-api](https://github.com/codepvg/leetcode-api) – API used to fetch user data from LeetCode
- [lc-backend](https://github.com/codepvg/lc-backend) – Backend service for storing and managing leaderboard data

These repositories handle the core data processing and integration logic for the leaderboard.

> 

Reviewer: ```suggestion

## Related Repositories

This project relies on external services for fetching and processing LeetCode data.
Instead of duplicating details here, refer to the following repositories:

- [leetcode-api](https://github.com/codepvg/leetcode-api) – API used to fetch user data from LeetCode
- [lc-backend](https://github.com/codepvg/lc-backend) – Backend service for storing and managing leaderboard data
- [frontend-uptime-monitor](https://github.com/codepvg/frontend-uptime-monitor) – Pinger service to monitor frontend server uptime
- [backend-uptime-monitor](https://github.com/codepvg/backend-uptime-monitor) – Pinger service to monitor backend server uptime

These repositories handle the core data processing and integration logic for the leaderboard.

```

---

## Project Structure

```
leetcode-ranking/
│── frontend/        # UI (HTML, CSS, JS)
│── data/            # Stored user data
│── scripts/         # Automation scripts
│── server.js        # Express server
│── package.json
```

---

## How to Run Locally

### 1. Clone the repository

> 

Reviewer: ```suggestion
### 1. Fork and clone the repository

First, fork the repository to your GitHub account. Then clone it locally:
```

```bash
git clone https://github.com/YOUR-USERNAME/leetcode-ranking.git
cd leetcode-ranking
```

### 2. Install dependencies

`npm install`
 cd frontend

> `npm install`

Reviewer: ```suggestion
```

### 3. Run the project

`npm run dev`

or

> node server.js

Reviewer: ```suggestion
`npm run dev`
or
`node start`
```

## Usage

1. Open the registration page  
2. Enter your name and LeetCode username  
3. Submit the form  
4. View your ranking on the leaderboard after 

>    the next sync

Reviewer: ```suggestion
4. View your ranking on the leaderboard after the next sync
```
